### PR TITLE
fix: Create next attempt after first attempt to initialize exponential backoff settings.

### PR DESCRIPTION
**Pull request by egreco12**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Review by yirutang — APPROVED** [review]

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java [hunk 1]
```diff
@@ -971,13 +971,15 @@ private Boolean retryOnRetryableError(Code errorCode, AppendRequestAndResponse r
...
         if (this.retrySettings != null && errorCode == Code.RESOURCE_EXHAUSTED) {
-          // Trigger exponential backoff in append loop when request is resent for quota errors
-          if (requestWrapper.attemptSettings == null) {
-            requestWrapper.attemptSettings = requestWrapper.retryAlgorithm.createFirstAttempt();
-          } else {
-            requestWrapper.attemptSettings =
-                requestWrapper.retryAlgorithm.createNextAttempt(requestWrapper.attemptSettings);
-          }
+          // Trigger exponential backoff in append loop when request is resent for quota errors.
+          // createNextAttempt correctly initiales the retry delay; createfirstAttempt does not
```
Comment (agrawal-siddharth) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java:975]: initiales ==> initializes

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java [hunk 2]
```diff
@@ -971,13 +971,15 @@ private Boolean retryOnRetryableError(Code errorCode, AppendRequestAndResponse r
...
+          // include a positive delay, just 0.
+          requestWrapper.attemptSettings =
+              requestWrapper.attemptSettings == null
+                  ? requestWrapper.retryAlgorithm.createFirstAttempt()
+                  : requestWrapper.attemptSettings;
+          requestWrapper.attemptSettings =
```
Comment (agrawal-siddharth) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java:981]: this line can be removed

Comment (agrawal-siddharth) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java:981]: What I meant to say is that lines 977 - 982 can be replaced with:

requestWrapper.attemptSettings = requestWrapper.retryAlgorithm.createNextAttempt(
requestWrapper.attemptSettings == null
                  ? requestWrapper.retryAlgorithm.createFirstAttempt()
                  : requestWrapper.attemptSettings
);

Comment (egreco12) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java]: Fixed

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java (hunk 1 above)
Comment (egreco12) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java]: Done

**Review by agrawal-siddharth — APPROVED** [review]

**Pull request merged**